### DISCUSSION TITLE
Help with permissions for file system delete

### DISCUSSION THREAD

**Discussion by chadwixk** [photoprism/photoprism#4150] [Q&A]
[PhotoPrism® CE]
Build [221118-e58fee0]

I am running photo-prism on my TrueNas server and am looking to use the Photo-Prism(PP) gui to do cleanup of unwanted photos, screenshots, etc.

PP is set to index originals. I have FolderSync android app on my wife's and my phones to push new photos to the TrueNas server.

I am able to multi-select unwanted photos, archive them, then delete them from PP. However, it is erroring out when deleting the original file. I assume this is due to permissions. I tried [the suggestion](https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions) of removing permissions for a test folder `sudo chmod -R a+rwX [folder]` but it still errored upon delete.

I'm not a linux guy, so please treat me like I'm 5 years old on linux commands :)

I turned on debug logs, but they are not that detailed. Not sure if there is a Log Level setting to give more insights?

Log:

<img width="664" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/b05e279f-c4ea-4e24-b313-3fc3279bfed5">

Error:

<img width="340" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/25ede3a8-67ff-4824-8f73-dea936f2a765">

Folder Permissions:

<img width="367" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/38ae80a6-9a7d-4e51-b8b2-0c7c59085ffc">

File Permissions:

<img width="533" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/a557eb76-90e0-4f06-a86f-671db7a66d32">

Extended Permissions:
<img width="649" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/c020325d-481e-4a4b-ac0a-c0e59acaf130">

I tried setting the log level to TRACE. I'm looking at the `messages` file in the directory : `iocage/jails/photo-prism/root/var/log` (not sure this is the correct one to view??)

And it doesn't really provide any further details on why the delete failed:
<img width="1394" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/2b2c914c-d28e-4a92-bccb-5b20984e5df1">

**Comment by graciousgrey** [comment]
I am not sure how TrueNAS handles volume mounts but is it possible that the originals folder is mounted with the :ro (read only) flag ?

**Reply by chadwixk** [reply]
The permissions on the mount are set to allow full control for everyone. 
<img width="442" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/ae1440a7-4944-43c6-bd69-e7914ff6d729">

Is there a way to get more information about why the delete failed? I set debug level to trace, but the error is still generic:
<img width="135" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/7ebaef6f-a8d5-49e5-a937-4427109ff00e">

<img width="600" alt="image" src="https://github.com/photoprism/photoprism/assets/5115308/3394ac75-8860-4c61-84d4-3854c5696a88">
